_overviews/core/programming-collections-generically.md

```diff
@@ -0,0 +1,320 @@
+      val lazyElements =
+        LazyList.unfold(()) { _ =>
+          if (Random.nextInt(100) < 10) None
+          else Some((genA.get, ()))
+        }
+      factory.fromSpecific(lazyElements)
+    }
+  }
+~~~
+
+The implementation uses a lazy source collection of a random size (`lazyElements`).
+Then it calls the `fromSpecific` method of the `Factory` to build the collection
+expected by the user.
+
+Here is an example of use of `collection`:
+
+~~~
+scala> collection[List, Int].get
+res0: List[Int] = List(606179450, -1479909815, 2107368132, 332900044, 1833159330, -406467525, 646515139, -575698977, -784473478, -1663770602)
+
+scala> collection[LazyList, Boolean].get
+res1: LazyList[Boolean] = LazyList(_, ?)
+
+scala> collection[Set, Int].get
+res2: Set[Int] = HashSet(-1775377531, -1376640531, -1009522404, 526943297, 1431886606, -1486861391)
+~~~
+
+## Transforming any collection
+
+In this section we will see how we can implement an `intersperse` operation that can be applied to
+any sequence and returns a sequence with a new element inserted between each element of the
+source sequence.
+
+Transforming collections consists in both consuming and producing collections. This is achieved by
+combining the techniques described in the previous sections. We can already start hacking with
+something like the following:
+
+~~~ scala
+import scala.collection.{ AbstractIterator, AbstractView, Factory, SeqOps }
+import scala.collection.generic.IsSeqLike
+
+class IntersperseOperation[A](seqOps: SeqOps[A, Iterable, _]) {
+  def intersperse[B >: A, That](sep: B)(implicit factory: Factory[B, That]): That =
+    factory.fromSpecific(new AbstractView[B] {
+      def iterator = new AbstractIterator[B] {
+        val it = seqOps.iterator
+        var intersperseNext = false
+        def hasNext = intersperseNext || it.hasNext
+        def next() = {
+          val elem = if (intersperseNext) sep else it.next()
+          intersperseNext = !intersperseNext && it.hasNext
+          elem
+        }
+      }
+    })
+}
+
+implicit def IntersperseOperation[Repr](coll: Repr)(implicit seq: IsSeqLike[Repr]): IntersperseOperation[seq.A] =
+  new IntersperseOperation(seq(coll))
+~~~
+
+However, if we try it we get the following behaviour:
+
+~~~
+scala> List(1, 2, 3).intersperse(0)
+res0: Array[Int] = Array(1, 0, 2, 0, 3)
+~~~
+
+We get back an `Array` although the source collection was a `List`! Indeed, there is
+nothing that constrains the result type of `interseprse` to depend on the receiver type.
+The following expressions show the behavior we expect:
+
+~~~ scala
+List(1, 2, 3).intersperse(0) == List(1, 0, 2, 0, 3)
+"foo".intersperse(' ') == "f o o"
+~~~
+
+When we call it on a `List`, we want to get back another `List`, and when we call it on
+a `String` we want to get back another `String`, and so on.
+
+To produce a collection whose type depend on a source collection, we have to use
+`scala.collection.BuildFrom` (formerly `CanBuildFrom`) instead of `Factory`.
```

Comment:
as always in this collections work, anytime `BuildFrom` appears, the snarky or skeptical reader will yell (silently, or perhaps on Twitter) "I thought you got rid of `CanBuildFrom`!

idk, but it might be worth expanding the parenthetical to a separate sentence like: "`BuildFrom` resembles the old collections API's `CanBuildFrom`, but appears much more rarely." idk.

Reply:
it's a bit of a tangent, but I wonder if the naming of `Factory` and `BuildFrom` could be more closely aligned with each other, since they are so closely related to each other. maybe it's too late.